webpack/move_to_pf/test-utils/testHelpers.js

```diff
@@ -53,11 +53,23 @@ export const testActionSnapshot = async (runAction) => {
 
 /**
  * Test actions with fixtures and snapshots
- * @param  {Object} fixtures key=fixture description, value=action runner function
+ * @param  {Object} fixtures  key=fixture description,
+ *                            value=action runner function or
+ *                                  object({
+ *                                    action = action runner function,
+ *                                    test = other function to run after the action
+ *                                  })
  */
 export const testActionSnapshotWithFixtures = fixtures =>
   Object.entries(fixtures).forEach(([description, runAction]) =>
-    it(description, () => testActionSnapshot(runAction)));
+    it(description, async () => {
+      if (typeof runAction === 'function') {
+        await testActionSnapshot(runAction);
+      } else {
+        await testActionSnapshot(runAction.action);
+        runAction.test();
+      }
+    }));
```

Comment:
I forgot to mention I took this from @sharvit's commit here https://github.com/Katello/katello/pull/7717/commits/6867ab12fa5eea2235ecb063bc580d213ac95f93

@sharvit let me know if you have any issues with this, I would like to use the functions here.

I would have to look more into it, do you see anything in thunk that would work similarly?

Reply:
@johnpmitsch I'm mostly just confused about it. Could you offer a bit of an explanation?

Reply:
So, the reason I mentioned thunk is because with normal redux, an action is always a JS object; with thunk, an action can be a function or a JS object. I think it's clear how snapshots can be used to test JS objects (it's just an expected JSON structure), but for functions, it looks like we're providing some params to the action function and expecting a certain API response.

No need to convert it; thanks for the explanation - I think I understand it much better now. Thunk *should* provide a helper for this if they don't. ;)